core/unit_test/TestRangePolicyConstructors.hpp

```diff
@@ -128,6 +128,8 @@ TEST(TEST_CATEGORY_DEATH, range_policy_implicitly_converted_bounds) {
   using IntIndexType  = Kokkos::IndexType<int>;
   using UIntPolicy    = Kokkos::RangePolicy<TEST_EXECSPACE, UIntIndexType>;
   using IntPolicy     = Kokkos::RangePolicy<TEST_EXECSPACE, IntIndexType>;
+  using IntAtomicDataElementType =
+      Kokkos::Impl::AtomicDataElement<Kokkos::ViewTraits<int>>;
```

Comment:
Please define a type that is convertible to the range policy integral type but not constructible from it instead.
It will improve readability and avoid introducing an unnecessary dependency with implementation details from other facilities in the codebase.

Reply:
Replaced to use a simple struct that is convertible to rangepolicy index_type instead.